Formula/node.rb

```diff
@@ -56,30 +56,11 @@ def install
 
     if build.with? "npm"
       resource("npm").stage buildpath/"npm_install"
-
-      # make sure npm can find node
-      ENV.prepend_path "PATH", bin
```

Comment:
Looking at the output of you gist:
* The first `yarn` `node-gyp` error seems unrelated to anything in this PR or the propsed `Language::node` changes
* I still think that mixing `yarn` and `npm` install commands in such a way is not a good idea:
    * While npm install grunt it will find a dep tree in the node_module folder, which it considers to be not optimal (in npm opinion). So npm will not only add grunt (and grunt's deps) into the node_module folder, but it will also reorganize the dep tree to be in a npm@5 style and moves modules around. It could also be possible that some change in npm@5 caused an incompatiblity with the yarn dep tree resulting in this error.